### ISSUE TITLE
API config for bodyparser doesn't respect extended:true

### ISSUE THREAD

**Issue by rdylina** [vercel/next.js#46107]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home
    Binaries:
      Node: 18.12.1
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.1.7-canary.21
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### Which area(s) of Next.js are affected? (leave empty if unsure)

Middleware / Edge (API routes, runtime), TypeScript

### Link to the code that reproduces this issue

https://github.com/rdylina/bodyparser-bug

### To Reproduce

set api config as follows

When passing deeply nested JSON to a NextJS API you need to enable extended:true in the bodyparser config otherwise nested objects will just be discarded and substituted for empty arrays or objects. setting extended to true in the body parser configuration as below is not respected and as such nested data continues to be truncated. 

export const config = {
  api: {
    bodyParser: {
      sizeLimit: '10mb',
      extended: true,
    },
  },
}

pass a deeply nested object via fetch.

### Describe the Bug

When passing deeply nested JSON to a NextJS API you need to enable extended:true in the bodyparser config otherwise nested objects will just be discarded and substituted for empty arrays or objects. setting extended to true in the body parser configuration as below is not respected and as such nested data continues to be truncated. 

export const config = {
  api: {
    bodyParser: {
      sizeLimit: '10mb',
      extended: true,
    },
  },
}

### Expected Behavior

I wold expect the full non-truncated JSON object to be available on API side after setting extended:true

### Which browser are you using? (if relevant)

Chrome Version 110.0.5481.104 (Official Build) (64-bit)

### How are you deploying your application? (if relevant)

localhost

**Comment by wwwdepot** [comment]
https://github.com/vercel/next.js/blob/0302576239231e5b97775c6b43d8a02ab291d514/packages/next/src/server/api-utils/node.ts#L474-L481

There's no such config `api.bodyParser.extended`. 

You can do is disable Next.js built-in body parser and use your own.

**Comment by rdylina** [comment]
Thats correct. But there should be a parameter for it. That's why I opened the issue.

And fyi, even with a custom body parser the data still gets truncated somewhere along the pipe by Next before it gets to the custom parser.

Easiest solution is just to match the standard and add the configuration parameter for extended.

Best,
Robert Dylina
Branch Manager NMLS#967879
Talk/Text: 209-233-0121
________________________________

**Comment by wwwdepot** [comment]
I see what you meant the body is being truncated. You're looking at `address: [Object]` in console.log & you think the body parsing went wrong. 

My solution is:
1. Instead of using `console.log`, use `console.dir(data, {depth:null})`
2. or use `console.log(JSON.stringify(data, null, 4))`
3. or return the data object as response so you can inspect it in the Network tab

**Comment by rdylina** [comment]
@wwwdepot You're absolutely correct. It does log fully to the console with that. The interesting thing that led me to even console.log the request at all was because nested values that I knew had values were coming back undefined when I tried to access them. I'll dig deeper.
